python/run-tests.py

```diff
@@ -99,8 +99,60 @@ def run_python_style_checks(root_dir):
     run_cmd([os.path.join(root_dir, "dev", "lint-python")], stream_output=True)
 
 
+def run_pypi_packaging_tests(root_dir):
+    print("##### Running PyPi Packaging tests #####")
+
+    version = '0.0.0'
+    with open(os.path.join(root_dir, "version.sbt")) as fd:
+        version = fd.readline().split('"')[1]
+
+    # uninstall packages if they exist
+    run_cmd(["pip3", "uninstall", "--yes", "delta-spark", "pyspark"], stream_output=True)
+
+    print("### Clearing Delta artifacts from ivy2 and mvn cache")
+    run_cmd(["rm", "-rf", "~/.ivy2/cache/io.delta/"], stream_output=True)
+    run_cmd(["rm", "-rf", "~/.m2/repository/io/delta/"], stream_output=True)
+
+    install_cmd = ["pip3", "install", "wheel", "twine", "setuptools", "--upgrade"]
+    print("### Executing:" + " ".join(install_cmd))
+    run_cmd(install_cmd, stream_output=True)
+
+    dist_dir = path.join(root_dir, "dist")
+
+    print("### Deleting `dist` directory if it exists")
+    delete_if_exists(dist_dir)
+
+    gen_artifacts_cmd_1 = ["python3", "setup.py", "bdist_wheel"]
+    print("### Executing: " + " ".join(gen_artifacts_cmd_1))
+    run_cmd(gen_artifacts_cmd_1, stream_output=True, stderr=open('/dev/null', 'w'))
+
+    gen_artifacts_cmd_2 = ["python3", "setup.py", "sdist"]
+    print("### Executing: " + " ".join(gen_artifacts_cmd_2))
+    run_cmd(gen_artifacts_cmd_2, stream_output=True)
+
+    # we need, for example, 1.1.0_SNAPSHOT not 1.1.0-SNAPSHOT
+    version_formatted = version.replace("-", "_")
+    delta_whl_name = "delta_spark-" + version_formatted + "-py3-none-any.whl"
+
+    install_whl_cmd = ["pip3", "install", path.join(dist_dir, delta_whl_name)]
+    print("### Executing: " + " ".join(install_whl_cmd))
+    run_cmd(install_whl_cmd, stream_output=True)
+
+    # run test python file directly with python and not with spark-submit
+    test_file = path.join(root_dir, path.join("examples", "python", "using_with_pip.py"))
```

Comment:
i am not sure how this is actually working. you are not really installing the maven artifact generated by current java/scala code. So this is probably running against the latest release maven artifacts. 

[This is the code](https://github.com/delta-io/delta/blob/ef76d4dc45748b2d3f7781555a740fe8954fde62/python/delta/pip_utils.py#L55) that is figuring out what version of delta to use when configuring sparksession to automatically download the delta maven artifact. my  guess is that this is still using delta 1.0 as that is the latest released version.

this is not right.. because our goal is to test the packaging with everything in the current code.  I think  you have to figure out a way to override this behavior. one way is to add an env var in the above linked code that specifies (i.e., 
 overrides importlib_metadata.version) which version to use for configuring spark session